sync/http/component.go

```diff
@@ -121,3 +122,149 @@ func (c *Component) createHTTPServer() *http.Server {
 		Handler:      routerAfterMiddleware,
 	}
 }
+
+// propSetMsg is used to log successes when using
+// the HTTP Builder setters, in a uniform way
+const propSetMsg = "Setting property '%v' for '%v'"
+
+// Builder gathers all required and optional properties, in order
+// to construct an HTTP component
+type Builder struct {
+	Component
```

Comment:
Since we have a builder now we can unexport the component.
I would not use structure embedding here since the builder "inherits" all fields and functions...

Reply:
@mantzas I agree with de-embedding the `Component` from the builder.

I'm not sure I understood the first point though. Do you suggest we unexport `Component` --> `component`? Then, both `Create()` and `New()` which are exported functions would return an unexported result, which is unwieldy, right? What am I missing?

Reply:
We have 2 options here:
- Leave it as is, which allows the developer to construct the component without using the builder, which defeats the purpose to have a builder
- Return the component interface and make the component unexported, which violates the "accept interfaces, return structs" guideline
I am ok with going with the first option...

Reply:
Do you feel okay with having the two build models co-existing for some time?

We could gradually introduce the builder as the new standard, by updating examples, opening PRs in other projects, etc. But, are we ultimately deprecating the old way of building http components (a breaking change)..?